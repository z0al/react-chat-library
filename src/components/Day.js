// @ts-check

// Packages
import React from 'react';
import { format } from 'date-fns';

/**
 * @typedef Props
 * @property {string} date
 * @property {string} [format]
 * @property {object} [containerStyle]
 * @property {object} [textStyle]
 * @param {Props} props
 */
const Day = props => (
	<div style={props.containerStyle}>
		<span style={props.textStyle}>{format(props.date, props.format)}</span>
		<style jsx>{`
			div {
				display: flex;
				justify-content: center;
				margin: 15px 0;
			}
			span {
				display: inline-block;
				font-size: 0.8em;
				padding: 0.2em 0.4em;
				text-align: center;
				color: #b2b2b2;
				font-weight: 600;
				text-transform: uppercase;
			}
		`}</style>
	</div>
);

/** @type Props */
Day.defaultProps = {
	date: '',
	format: 'MMM D, YYYY',
	containerStyle: {},
	textStyle: {}
};

export default Day;
