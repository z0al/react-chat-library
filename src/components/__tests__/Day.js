import React from 'react';
import renderer from 'react-test-renderer';

import Day from '../Day';

test('should match the snapshot', () => {
	const tree = renderer.create(<Day />).toJSON();
	expect(tree).toMatchSnapshot();
});
