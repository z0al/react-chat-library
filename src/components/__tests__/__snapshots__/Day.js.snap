// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="jsx-2228099478"
  style={Object {}}
>
  <span
    className="jsx-2228099478"
    style={Object {}}
  >
    Invalid Date
  </span>
</div>
`;
